### PR TITLE
Mesh-653 Staking updated after upgrade to v2

### PR DESCRIPTION
**Pull request by vkandy**
1. Compliance status updated at genesis time to accounts adding at genesis
2. Compliance status for accounts in tests
3. Added missing logic from old staking module. Only validators that have passed compliance status are considered for validation in the next era.

### CODE REVIEW

File: custom/staking/src/lib.rs [hunk 1]
```diff
@@ -807,6 +810,16 @@ decl_storage! {
...
+                        ).ok();
+                        <Module<T>>::add_potential_validator(
+                            // TODO: change origin to committee
+                            frame_system::RawOrigin::Root.into(),
+                            stash.clone()
+                        ).ok();
+                        <Module<T>>::compliance_passed(
+                            // TODO: change origin to committee
+                            frame_system::RawOrigin::Root.into(),
+                            stash.clone()
```
Comment (satyamakgec) [inline custom/staking/src/lib.rs]: isn't `controller.clone()` instead of `stash.clone()` ?

Comment (vkandy) [reply custom/staking/src/lib.rs]: No, it's the stash acccounts. Tests are written that way as well.

Comment (satyamakgec) [reply custom/staking/src/lib.rs]: ahh make sense after checking the actual code, I was confused with signature of `add_potential_validator()` function i.e 
```
fn add_potential_validator(origin, controller: T::AccountId) 
```
I think it is better to change the `controller` name to `stash` for consistency.  same applies for `remove_validator` ,`compliance_failed` & `compliance_passed`

Comment (vkandy) [reply custom/staking/src/lib.rs]: Yeah - the arg name should have been `validator`. Renamed args.

File: custom/staking/src/lib.rs [hunk 2]
```diff
@@ -660,6 +660,9 @@ pub trait Trait: frame_system::Trait + pallet_babe::Trait {
 
     /// Required origin for changing compliance status (can always be Root).
     type RequiredComplianceOrigin: EnsureOrigin<Self::Origin>;
+
+    /// Required origin for changing validator commission.
+    type RequiredCommissionOrigin: EnsureOrigin<Self::Origin>;
```
Comment (maxsam4) [inline custom/staking/src/lib.rs]: Should this be a part of this PR?

**Pull request merged**